numba/core/dispatcher.py

```diff
@@ -679,6 +687,7 @@ def typeof_pyval(self, val):
         else:
             if tp is None:
                 tp = types.pyobject
+        self._types_active_call = tp
```

Comment:
I think this will have to be a `self._types_active_call.append(tp)`.

Reply:
you're right, fixed!